### PR TITLE
Add HtmlRenderer.Test project, port applicable tests from PeachPDF

### PR DESCRIPTION
**Pull request by jhaygood86**
## Summary

Ports applicable tests from [PeachPDF](https://github.com/PeachyPrinting/PeachPDF)'s test suite (PeachPDF forked from HTML-Renderer and has since built out a much larger feature set) into three MSTest projects, routed by what each test actually exercises:

- `HtmlRenderer.Test` (new) — pure unit tests against the core library: CSS engine/parsing, DOM/layout utilities. Backed by a lightweight mock `RAdapter`/`RGraphics` test harness with no UI-framework dependency.
- `HtmlRenderer.IntegrationTest` — end-to-end layout/paint behavior tests, using a similar harness backed by the real WinForms adapter.
- `HtmlRenderer.PdfSharp.Test` — PDF-generation and PdfSharp-adapter tests.

This PR was opened when HTML-Renderer still had its old, much more limited hand-rolled CSS parser, so only ~41 of PeachPDF.Tests' ~483 files were portable at the time. **Since then, HTML-Renderer absorbed a full CSS engine port from PeachPDF/ExCSS** (`Source/HtmlRenderer/Core/CssEngine/`, a near-1:1 structural clone of PeachPDF's own CSS engine), which changed the picture substantially — most of `PeachPDF.Tests/CSS/` tests the CSS *parsing*/CSSOM layer in isolation, and that layer is now close to feature-complete. This PR was updated to:

1. Fix the ~20 files ported before the CSS engine landed, which broke against the new APIs (adapter signature drift, removed legacy parsing methods, a proprietary `corner-radius` mechanism replaced by real `border-radius`).
2. Re-verify all 111 previously `[Ignore("not yet spec compliant")]` tests against the current codebase — 28 now genuinely pass (mostly `CssLength` unit-conversion coverage, several layout edge cases, and a couple of table `visibility:collapse`/`vertical-align` cases) and are un-ignored; the rest still fail for their originally-documented reasons, which the CSS engine port never touched (it replaced parsing, not the box/layout/paint engine).
3. Re-triage and port the remaining ~86 files in `PeachPDF.Tests/CSS/` against the new engine — the vast majority are now portable, since they test parsing/CSSOM in isolation rather than rendered behavior.

**What's still excluded, and why:** `PeachPDF.Tests/Html/`, `Integration/`, `PdfSharpCore/`, `Svg/`, and misc folders (~415 files) have not been re-triaged against the new engine in this pass — those test *rendered* behavior (layout/paint), and CSS-parsing support doesn't imply layout support. Confirmed directly: flexbox, grid, `transform`, `box-shadow`, and animations all parse correctly now but have **zero** consumption anywhere in the layout/paint engine (`CssLayoutEngine.cs` has no "flex"/"grid" references, no paint handler draws a shadow or applies a transform). `border-radius` is the one exception, genuinely painted end-to-end. Also still absent regardless of layer: WOFF/WOFF2 binary decoding, GCPM paged-media content functions (`leader()`, `target-counter()`, `running()`, etc.), SVG rendering, `:has()` with leading combinators, and font-variant granular sub-properties.

Where a ported test exercises a feature that *does* exist but isn't yet spec-compliant, it's still ported with its full original assertions intact — documenting the real target behavior — but marked `[Ignore("not yet spec compliant")]` with a doc comment citing the exact source evidence for the gap. This pass surfaced several new gaps this way: `gap`/`row-gap`/`column-gap` reject the `normal` keyword, `font-weight` only accepts the legacy CSS2.1 100-900 multiples (not the full `[1,1000]` range), `GridTemplate`/`GridTrackSize` have no value-equality, `@supports`/`@container` parse but are never consulted by the real per-box cascade, and cascade-layer precedence isn't implemented (rules in `@layer` are treated as ordinary unlayered rules).

Adds `InternalsVisibleTo` grants from `HtmlRenderer`/`HtmlRenderer.WinForms`/`HtmlRenderer.PdfSharp` to their respective test projects, matching the same grants PeachPDF's own `.csproj` already declares for `PeachPDF.Tests`. No behavioral changes to HTML-Renderer itself.

More tests will be ported incrementally as HTML-Renderer's layout/paint engine catches up to what the CSS engine can already parse, and as the remaining PeachPDF.Tests folders get their own re-triage pass.

## Test plan

- [x] `dotnet build Source/HtmlRenderer.sln` (Release) — clean, 0 errors
- [x] `dotnet test` per project:
  - `HtmlRenderer.Test`: 2367 passed, 0 failed, 123 skipped (ignored, spec-compliance gaps)
  - `HtmlRenderer.IntegrationTest` (new tests): 92 passed, 0 failed, 74 skipped
  - `HtmlRenderer.PdfSharp.Test`: 15 passed, 0 failed
- [x] All 9 CI matrix jobs (Windows/Ubuntu/macOS × .NET 8/9/10) pass.

### CODE REVIEW

**Comment by eXpl0it3r** [comment]
Rebased onto master and removed Claude co-author attribution

File: Source/HtmlRenderer.PdfSharp/HtmlRenderer.PdfSharp.csproj [hunk 1]
```diff
@@ -29,6 +29,9 @@ Features and Benefits:
   <ItemGroup>
     <ProjectReference Include="..\HtmlRenderer\HtmlRenderer.csproj" />
   </ItemGroup>
+  <ItemGroup>
+    <InternalsVisibleTo Include="HtmlRenderer.PdfSharp.Test" />
+  </ItemGroup>
   <ItemGroup>
     <PackageReference Include="Microsoft.Win32.Registry" Version="5.0.0" />
```
Comment (Wolffy101) [inline Source/HtmlRenderer.PdfSharp/HtmlRenderer.PdfSharp.csproj]: Microsoft.Win32.Registry  is no longer used
Microsoft.Win32.Registry 不再使用了

**Review by eXpl0it3r — APPROVED** [review]
Rather verbose at times and quite a few ignored tests, but it's a good foundation to optimize on later

**Comment by eXpl0it3r** [comment]
Thank you for porting all these tests! 🎉
